docs/kb/function-tutorial-intro.md

```diff
@@ -0,0 +1,239 @@
+- **Dockerfile** contains the instructions for the Docker application on how to build the Docker image.  You most likely will not need to modify this file.
+- **README.md** is the default web page you see at [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack).  It is written in markdown language.
+- **function.yaml** contains the instructions to load the VEBA function into the Kubernetes cluster.  This file has two pieces of information that you may want to change:
+  1. The vCenter Event that triggers the function.  The default event is: **VmPoweredOffEvent** but this may be modified to suit your requirements.
+  2. The location of the Docker image to load into Kubernetes.  If the PowerShell code will not be modified, you can leave the default image as is.  If you modify the PowerShell code in the handler file, you will need to build a new Docker image and specify the name/location in this file.
+- **handler.ps1** contains the PowerShell code that runs when the specified vCenter event triggers it.  The handler.ps1 file is injected into the Docker image which must be rebuilt every time the code changes.
+- **slack_secret.json** contains the Slack key for your Slack channel.  This file will be used to create a Kubernetes secret that the function will read.  Most functions will include a secret file that contains: usernames/passwords, keys, or other sensitive information that will be used to create a Kubernetes secret.
+
+**What are Kubernetes secrets?**  The VEBA appliance runs a Kubernetes cluster on top of the Photon OS.  This allows the VEBA functions to be packaged as containers that are then deployed into the Kubernetes cluster.  A Kubernetes secret is an object that contains a small amount of sensitive data that can be read by a Kubernets pod (pods are made up of containers).  Secrets allow sensitive or confidential data to be packaged separate from a container's application code.
+
+Also note the link in the README for the available video tutorial which may be of help: [video tutorial](https://youtu.be/vLyeZrm0jk4)
+
+
+## Installing required tools on your workstation
+You will need to install and configure/use three tools:
+- git - used to clone the VEBA github repository down to your workstation (including the function code we want to use) so that you have a local copy to modify/test/deploy.
+- Docker - if you modify the function's code, you will use Docker to:
+  - create a new Docker image
+  - push the image to the Docker registry so that it is available to be pulled down by the Kubernetes cluster.
+  - test the function locally on your workstation without the need for a deployed veba appliance.
+- kubectl - command line interface to the Kubernetes cluster.  This command is used to deploy the function and secret to the Kubernetes cluster.  It is also used in troubleshooting the VEBA appliance.
```

Comment:
```suggestion
- `git` - used to clone the VEBA github repository down to your workstation (including the function code we want to use) so that you have a local copy to modify/test/deploy.
- Docker - if you modify the function's code, you will use Docker to:
  - create a new Docker image
  - push the image to the Docker registry so that it is available to be pulled down by the Kubernetes cluster.
  - test the function locally on your workstation without the need for a deployed veba appliance.
- `kubectl` - command line interface to the Kubernetes cluster.  This command is used to deploy the function and secret to the Kubernetes cluster.  It is also used in troubleshooting the VEBA appliance.
```

Reply:
Please change these accordingly in all docs